projects/ngx-prx-styleguide/src/lib/tooltip/tooltip.directive.ts

```diff
@@ -0,0 +1,61 @@
+// From https://github.com/pleerock/ngx-tooltip#readme not maintained, recommended to fork
+import {
+  Directive,
+  HostListener,
+  ComponentRef,
+  ViewContainerRef,
+  Input,
+  ComponentFactoryResolver
+} from '@angular/core';
+import { TooltipContentComponent } from './tooltip-content.component';
+
+@Directive({
+  selector: '[tooltip]'
+})
+export class TooltipDirective {
+  private tooltip: ComponentRef<TooltipContentComponent>;
+  private visible: boolean;
+
+  constructor(private viewContainerRef: ViewContainerRef, private resolver: ComponentFactoryResolver) {}
+
+  @Input('tooltip')
+  content: string;
+
+  @Input()
+  tooltipDisabled: boolean;
+
+  @Input()
+  tooltipAnimation = true;
+
+  @Input()
+  tooltipPlacement: 'top' | 'bottom' | 'left' | 'right' = 'bottom';
+
+  @HostListener('focusin')
+  @HostListener('mouseenter')
+  show(): void {
```

Comment:
uses standard HostListener stuff with ComponentFactoryResolver to create a component that actually shows the content string

Reply:
and not a bad small example for creating ad-hoc components 😸